tools/osbuilder/rootfs-builder/rootfs.sh

```diff
@@ -667,6 +667,20 @@ EOF
 		chmod g+rx,o+x "${ROOTFS_DIR}"
 	fi
 
+	# Restricting access to agent endpoints using agent-config.toml is expected to
+	# be deprecated in the main branch. Therefore, in preparation of merging this
+	# script with its main branch version, install default settings for main branch's
+	# kata-opa service. coco-default.rego blocks access to the same kata agent
+	# endpoints that are blocked by agent-config.toml. For additional information,
+	# search for "default-policy.rego" in main branch's rootfs.sh.
+	local kata_opa_in_dir="${script_dir}/../../../src/kata-opa"
+	local opa_settings_dir="/etc/kata-opa"
+	local policy_file="coco-default.rego"
+	local policy_dir="${ROOTFS_DIR}/${opa_settings_dir}"
+	mkdir -p "${policy_dir}"
+	install -D -o root -g root -m 0644 "${kata_opa_in_dir}/${policy_file}" -T "${policy_dir}/${policy_file}"
+	ln -sf "${policy_file}" "${policy_dir}/default-policy.rego"
```

Comment:
So just to double check I understand this, as part of the merge to main for endpoint restriction, we'll want to take over the coco-deafult.rego file and then for the TEE_PLATFORM="SEV/SNP" (at least and probably SE & TDX too), we'll want to update the existing logic in root.sh to link from `coco-default.rego` instead of `allow-all.rego`?

Reply:
Good to know about TEE_PLATFORM - I wasn't aware of it!

When we'll merge the two versions of rootfs.sh, I hope git will complain about a merge conflict - since these few lines of the script are almost identical but not quite across the two branches. That conflict will force us to think about which default policy we'll want on which platforms:

- If we were to merge today, I would enable the new coco-default.rego as default policy for any TEE_PLATFORM, and continue to use all-allowed.rego for all non-TEE platforms that support Policy. 

- But by the time of the merge it's possible that we'll come up with better default policies than coco-default.rego. The current coco-default.rego is safe enough for the short term but it's not flexible enough for the future.

Reply:
Just to clarify - `TEE_PLATFORM` hasn't been written yet, but is the current best plan we have for getting the attestation-agent variants in for the different architectures.

> If we were to merge today, I would enable the new coco-default.rego as default policy for any TEE_PLATFORM, and continue to use all-allowed.rego for all non-TEE platforms that support Policy.
>
> But by the time of the merge it's possible that we'll come up with better default policies than coco-default.rego. The current coco-default.rego is safe enough for the short term but it's not flexible enough for the future.

Great, that matches my expectation!